build/build.c

```diff
@@ -213,24 +278,58 @@ static rpmRC buildSpec(BTA_t buildArgs, rpmSpec spec, int what)
     } else {
 	int didBuild = (what & (RPMBUILD_PREP|RPMBUILD_BUILD|RPMBUILD_INSTALL));
 
+	if (!spec->buildrequires && (what & RPMBUILD_PACKAGESOURCE) &&
+	    !(what & (RPMBUILD_BUILD|RPMBUILD_INSTALL|RPMBUILD_PACKAGEBINARY))){
+		/* don't run prep if not needed for source build */
+		/* with(out) dynamic build requires*/
+	    what &= ~(RPMBUILD_PREP);
+	}
+
+	if ((what & RPMBUILD_CHECKBUILDREQUIRES) &&
+	    (rc = doCheckBuildRequires(ts, spec, test)))
+		goto exit;
+
 	if ((what & RPMBUILD_PREP) &&
 	    (rc = doScript(spec, RPMBUILD_PREP, "%prep",
-			   getStringBuf(spec->prep), test)))
+			   getStringBuf(spec->prep), NULL, test)))
 		goto exit;
 
+	if (what & RPMBUILD_BUILDREQUIRES)
+            rc = doBuildRequires(spec, test);
+	if ((what & RPMBUILD_CHECKBUILDREQUIRES) &&
+	    (rc == RPMRC_MISSINGBUILDREQUIRES))
+	    rc = doCheckBuildRequires(ts, spec, test);
```

Comment:
The condition and if-block are on the same indentation level here, please fix.

This whole doScript() related if-section is quite a mess to begin with and violates rpm general style, so an alternative to indenting the conditional is indenting the if-block deeper, as is done with the goto exit's below. Readability is far more important than following style to the letter, and consistency with surroundings helps readability too.